compiler/pipes/instantiate-generics-and-lambdas.cpp

```diff
@@ -48,15 +50,22 @@ class InstantiateGenericFunctionPass final : public FunctionPassBase {
           type_hint = phpdoc_replace_genericTs_with_reified(type_hint, instantiationTs);
         }
       }
+      // inside f<A>, replace `classof($o)` with "A" if `f<T>(T $o)` called as `f(expr_assumed_A)`
+      if (as_call->str_val == "classof" && as_call->size() == 1 && as_call->extra_type != op_ex_func_call_arrow) {
```

Comment:
Why do you check `as_call->extra_type != op_ex_func_call_arrow`?
Could you give an example where it is the case?

Reply:
Just any arrow call with such a method name :)
```
$obj->classof(...)
```
It's represented as `classof($obj, ...)` in AST with `extra_type = op_ex_func_call_arrow`